Sources/_SwiftSyntaxMacros/ExpressionMacro.swift

```diff
@@ -20,3 +24,54 @@ public protocol ExpressionMacro: Macro {
     _ macro: MacroExpansionExprSyntax, in context: MacroEvaluationContext
   ) -> MacroResult<ExprSyntax>
 }
+
+#if canImport(_CompilerPluginSupport)
+extension ExpressionMacro {
+  public static func _kind() -> _CompilerPluginKind {
+    .expressionMacro
+  }
+
+  public static func _rewrite(
+    targetModuleName: UnsafePointer<UInt8>,
+    targetModuleNameCount: Int,
+    filePath: UnsafePointer<UInt8>,
+    filePathCount: Int,
+    sourceFileText: UnsafePointer<UInt8>,
+    sourceFileTextCount: Int,
+    localSourceText: UnsafePointer<UInt8>,
+    localSourceTextCount: Int
+  ) -> (UnsafePointer<UInt8>?, count: Int) {
+    let targetModuleNameBuffer = UnsafeBufferPointer(
+      start: filePath, count: targetModuleNameCount)
+    let targetModuleName = String(
+      decoding: targetModuleNameBuffer, as: UTF8.self)
+    let filePathBuffer = UnsafeBufferPointer(
+      start: filePath, count: filePathCount)
+    let filePath = String(decoding: filePathBuffer, as: UTF8.self)
+    let sourceFileTextBuffer = UnsafeBufferPointer(
+      start: sourceFileText, count: sourceFileTextCount)
+    let sourceFileText = String(decoding: sourceFileTextBuffer, as: UTF8.self)
+    // TODO: `SourceLocationConverter.init(file:source:)` should be able to
```

Comment:
`SourceLocationConverter` has an initializer that takes a syntax. If you parse the whole source file first, then you can initialize `SourceLocationConverter` with the `SourceFileSyntax` you get back.